### PR TITLE
Parallel computing update

### PR DESCRIPTION
**Pull request by UnciaBit**
## Details

Since some of the additional sections were not finished, I've created a new branch called parallel-computing-stash which contains a complete version of this chapter, where all the comments made in #13 are addressed.

## Linked Issues

- #6

### CODE REVIEW

File: src/chapter4/openmp.md [hunk 1]
```diff
@@ -1 +1,101 @@
...
+)
+
+Here is an example of `#pragma`
+- The function starts with serial region
+- At the line `#pragma omp parallel`, a group of threads are spawned to create parallel region inside the bracket 
+- At the end of the bracket, the program goes back to serial computing 
+
+## Running "Hello World" on Multi-threads
+
+>If you're unsure about the difference between **multi-threading** and **multi-processing**, check the page [here](src/chapter4/multithreading.md)
```
Comment (jasparm) [inline src/chapter4/openmp.md:55]: change the link to a relative one, eg `./multithreading.md`

File: src/chapter4/challenges.md [hunk 2]
```diff
@@ -1 +1,61 @@
...
+5. Run parallel code and check the improved run time
+
+## Task 2 - Run task 1 on HPC cluster
+
+1.  Check the available partitions with `show_cluster`
+2.  Modify `RunHello.sh `
+3.  `sbatch RunHello.sh`
+4.  `cat slurm<>.out` and check the run time
+
+>You can also use strudel web to run the script without sbatch: https://beta.desktop.cvl.org.au/login
```
Comment (jasparm) [inline src/chapter4/challenges.md:20]: change the url to a link

File: src/chapter4/openmp.md [hunk 3]
```diff
@@ -1 +1,101 @@
...
+
+**User** is the amount of CPU time spent outside the kernel within the process
+
+**Sys** is the amount of CPU time spent in the kernel within the process. 
+**User** time + **Sys** time will tell you how much actual CPU time your process used. 
+
+## More Features of OpenMP
+
+- [Introduction to OpenMP](https://www.youtube.com/watch?v=iPb6OLhDEmM&list=PLLX-Q6B8xqZ8n8bwjGdzBJ25X2utwnoEG&index=11 )
+- [\#pragma omp parallel private](https://www.youtube.com/watch?v=dlrbD0mMMcQ&list=PLLX-Q6B8xqZ8n8bwjGdzBJ25X2utwnoEG&index=17)
```
Comment (jasparm) [inline src/chapter4/openmp.md:100]: these names are a bit weird, maybe change to properly describe what they are linking to

**Review by jasparm — COMMENTED** [review]
Generally looks good, just a few small nitpicks

File: src/chapter4/openmp.md (hunk 1 above)
Comment (UnciaBit) [reply src/chapter4/openmp.md]: Addressed

File: src/chapter4/challenges.md (hunk 2 above)
Comment (UnciaBit) [reply src/chapter4/challenges.md]: Addressed

File: src/chapter4/openmp.md (hunk 3 above)
Comment (UnciaBit) [reply src/chapter4/openmp.md]: Addressed

**Review by oraqlle — APPROVED** [review]
Done

**Pull request merged**